cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/connector/resource/AzureDatabaseResourceService.java

```diff
@@ -238,4 +244,76 @@ public ExternalDatabaseStatus getDatabaseServerStatus(AuthenticatedContext ac, D
     public String getDBStackTemplate() {
         return azureDatabaseTemplateProvider.getDBTemplateString();
     }
+
+    public void upgradeDatabaseServer(AuthenticatedContext authenticatedContext, DatabaseStack stack,
+            PersistenceNotifier persistenceNotifier, TargetMajorVersion targetMajorVersion) {
+
+        CloudContext cloudContext = authenticatedContext.getCloudContext();
+        Long stackId = cloudContext.getId();
+        Optional<CloudResource> databaseServer = persistenceRetriever.retrieveFirstByTypeAndStatusForStack(AZURE_DATABASE,
+                CommonStatus.CREATED, stackId);
+        Optional<CloudResource> privateEndpoint = persistenceRetriever.retrieveFirstByTypeAndStatusForStack(AZURE_PRIVATE_ENDPOINT,
+                CommonStatus.CREATED, stackId);
+
+        if (databaseServer.isPresent()) {
+            CloudResource databaseServerResource = databaseServer.get();
+            String databaseReference = databaseServerResource.getReference();
+            LOGGER.info("Azure database server has been found with the reference '{}'",
+                    databaseReference);
+            String stackName = azureUtils.getStackName(cloudContext);
+            String resourceGroupName = azureResourceGroupMetadataProvider.getResourceGroupName(cloudContext, stack);
+
+            stack.getDatabaseServer().putParameter(DB_VERSION, targetMajorVersion.getMajorVersion());
+            String template = azureDatabaseTemplateBuilder.build(cloudContext, stack);
+
+            AzureClient client = authenticatedContext.getParameter(AzureClient.class);
+            try {
+                privateEndpoint.ifPresent(resource -> deletePrivateEndpoint(client, resource, persistenceNotifier, cloudContext));
+                deleteDatabaseServer(client, databaseServerResource, persistenceNotifier, cloudContext);
+                deployDatabaseServer(stackName, resourceGroupName, template, client);
+            } catch (CloudException e) {
+                throw azureUtils.convertToCloudConnectorException(e, "Database stack upgrade");
+            } catch (Exception e) {
+                throw new CloudConnectorException(String.format("Error in upgrading database stack %s: %s", stackName, e.getMessage()), e);
+            } finally {
+                updateCloudResourcesInDeployment(persistenceNotifier, cloudContext, stackName, resourceGroupName, client);
+            }
+        } else {
+            String exceptionMessage = String.format("Azure database server cloud resource does not exist for stack %d!", stackId);
+            LOGGER.warn(exceptionMessage);
+            throw new CloudConnectorException(exceptionMessage);
```

Comment:
wouldn't this cause an issue if the pod is restarted after the db deleted but before it's created?